clang/include/clang/CIR/Dialect/IR/CIROpenCLAttrs.td

```diff
@@ -168,4 +168,19 @@ def OpenCLVersionAttr : CIR_Attr<"OpenCLVersion", "cl.version"> {
   let assemblyFormat = "`<` $major_version `,` $minor_version `>`";
 }
 
+
+//===----------------------------------------------------------------------===//
+// OpenCLKernelAttr
+//===----------------------------------------------------------------------===//
+
+def OpenCLKernelAttr : CIRUnitAttr<
+    "OpenCLKernel", "cl.kernel"> {
```

Comment:
Given that other language extensions also have kernels (with perhaps different keywords), have you thought about having something more generic, like `kernel(opencl)`? In the future that could also be `kernel(cuda)`, etc?

Reply:
To be more concrete here: we could have generic kernel attribute and have enum kinds for the flavor.

Reply:
It's also fine if this comes later but we need at least a `TODO` comment here to codify the intent.

Reply:
TODO attached. A enum in extra attribute might be troublesome to operate. I'll try adding some helper methods for `ExtraFuncAttr` first.

Reply:
Hmm good point, now that you added the TODO perhaps leave this to future to whenever other kernel flavors get added? I'll leave the decision to you, thanks for adding the TODO